arrow-flight/src/sql/client.rs

```
@@ -518,17 +520,36 @@ impl PreparedStatement<Channel> {
                 .await
                 .map_err(flight_error_to_arrow_error)?;
 
-            self.flight_sql_client
+            let result = self
+                .flight_sql_client
                 .do_put(stream::iter(flight_data))
                 .await?
-                .try_collect::<Vec<_>>()
+                .message()
                 .await
-                .map_err(status_to_arrow_error)?;
+                .map_err(status_to_arrow_error)?
+                .unwrap();
+            // Attempt to update the stored handle with any updated handle in the DoPut result.
+            // Not all servers support this, so ignore any errors when attempting to decode.
```

I think this might silently ignore legitimate errors and I think it would be a good way to distinguish between a server not sending a response vs sending an invalid response. 

If the server doesn't support sending back an updated handle, the put_results should be empty, right? So perhaps we could check for an empty response rather than ignoring the error?

---

I've updated the logic here to not ignore legitimate errors and instead only look for empty responses instead.